exporters/otlp/otlp.go

```diff
@@ -195,6 +195,36 @@ func (e *Exporter) dialToCollector() (*grpc.ClientConn, error) {
 	return grpc.DialContext(ctx, addr, dialOpts...)
 }
 
+// Shutdown closes all connections and releases resources currently being used
+// by the exporter. If the exporter is not started this does nothing.
+func (e *Exporter) Shutdown(ctx context.Context) {
+	e.mu.RLock()
+	cc := e.grpcClientConn
+	started := e.started
+	e.mu.RUnlock()
+
+	if !started {
+		return
+	}
+
+	if err := cc.Close(); err != nil {
+		// Nothing we can do about this failure here.
```

Comment:
Why not return `error` from Shutdown()?

Reply:
Yeah, I went back and forth with this one. I ended up not including it in the function signature because it doesn't seem like there will be anything in the calling stack that will be better able handle this error than this function. If the shutdown fails the caller cannot just re-run and to directly deal with the error it would have to know things like is it from file cleanup or network hiccups of which knowledge it will not have.

I'm open to alternative perspectives if you think we can provide a better API by accepting this. Thoughts?

Reply:
I had the same reaction, wondering why `Shutdown` doesn't return an `error`.

Does this `global.Handle` function terminate the program?

Reply:
@seh `global.Handle` just routes the error to the default (log the error) or user defined error [Handler](https://github.com/open-telemetry/opentelemetry-go/blob/master/error_handler.go#L18).

I'm fine changing the function signature based on the growing consensus (I didn't have a strong preference here). @open-telemetry/go-approvers are there any other opinions here?

Reply:
I asked because absent calling `panic` or `os.Exit`, such a `Handler` won't change the flow of control here. You're not exiting the function if there's an error detected; you proceed past where you call on the `global.Handle` function. Does it matter whether the function finishes it work, or would you return a non-nil `error` at this point?

Reply:
> I asked because absent calling `panic` or `os.Exit`, such a `Handler` won't change the flow of control here. You're not exiting the function if there's an error detected; you proceed past where you call on the `global.Handle` function. Does it matter whether the function finishes it work, or would you return a non-nil `error` at this point?

Ah, good point. You are correct that needs to change. I was definitely overlooking that. Thanks for the catch

Reply:
@seh digging deeper, I think the original was correct. If this errors, we still want to ensure the collector is shutdown (change state and clean the background connector.

Reply:
That's fine. My point was more about either moving the call to `global.Handle` to the end, so that you always complete the intended work here, or replacing that call with returning any non-nil error value at the end, rather than interrupting the function with the error handling.